scripts/calculator.js

```diff
@@ -0,0 +1,121 @@
+import { Expression } from "./expression.js";
+import { History } from "./history.js";
+
+class Calculator {
+    constructor (calculatorSelector, inputFieldSelector, buttonSelectors) {
+        // DOM Selectors
+        this.calculatorContainer = document.querySelector(calculatorSelector);
+        this.inputField = document.querySelector(inputFieldSelector);
+        this.buttons = document.querySelectorAll(buttonSelectors);
+
+        this.currentInput = "";
+        this.resultDisplayed = false;
+        this.history = [];
+
+        this.expression = new Expression();
+        this.h = new History(this.history, "history");
+        
+        this.init();
+    }
+
+    init () {
+        this.buttons.forEach(button => {
+            button.addEventListener("click", (e) => this.handleButtonClick(e));
+        });
+
+        document.addEventListener("keydown", (e) => this.handleKeyEvent(e));
+    }
+
+    // for update input field values
+    updateInputField (value) {
+        if (this.resultDisplayed && /[0-9.]/.test(value)) {
+            this.clearInputField();
+        }
+        this.resultDisplayed = false;
+        this.currentInput += value;
+        this.inputField.textContent = this.currentInput;
+        this.inputField.scrollTo(this.inputField.offsetWidth, 0);
+    }
+
+    // for clear eniter input field
+    clearInputField () {
+        this.currentInput = "";
+        this.inputField.textContent = this.currentInput;
+    }
+
+    // for remove only last charecter
+    removeLastCharacter () {
+        this.currentInput = this.currentInput.slice(0, -1);
+        this.inputField.textContent = this.currentInput;
+    }
+
+    // for handle mouse event
+    handleButtonClick (e) {
+        const value = e.target.closest("button").value;
+
```

Comment:
Perhaps we can add one early return if we don't find any button.
```suggestion
        const button = e.target.closest("button");
        if(!button) return;
        const value = button.value;

```

Reply:
Added early if we don't find button (https://github.com/krushna-simform/scientific-calculator/pull/3/commits/5c3476567a467a7080c3f0e458850eb9558b3e0f)